File: core/src/main/java/farayan/sabad/core/OnePlace/ProductBarcode/ProductBarcodeEntity.java
package farayan.sabad.core.OnePlace.ProductBarcode;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import farayan.sabad.core.OnePlace.product.ProductEntity;
import farayan.sabad.core.base.SabadEntityBase;

@DatabaseTable(tableName = ProductBarcodeSchema.ProductBarcodes)
public class ProductBarcodeEntity extends SabadEntityBase<ProductBarcodeEntity>
{

	@DatabaseField(columnName = ProductBarcodeSchema.BitmapResultPoints)
	public String BitmapResultPoints;

	@DatabaseField(columnName = ProductBarcodeSchema.BitmapScaleFactor)
	public int BitmapScaleFactor;

	@DatabaseField(columnName = ProductBarcodeSchema.BitmapFile)
	public String BitmapFile;

	@DatabaseField(columnName = ProductBarcodeSchema.Text)
	public String Text;

	@DatabaseField(columnName = ProductBarcodeSchema.Format)
	public BarcodeFormats Format;

	@DatabaseField(columnName = ProductBarcodeSchema.Product, foreign = true)
	public ProductEntity Product;

	public ProductBarcodeEntity() {

	}


	public ProductBarcodeEntity(String barcode, ProductEntity product) {
		Text = barcode;
		Product = product;
	}

	@Override
	public boolean NeedsRefresh() {
		return Text == null;
	}
}

File: app/src/main/java/farayan/sabad/model/product_barcode/ProductBarcodeEntity.kt
package farayan.sabad.model.product_barcode

import com.google.zxing.BarcodeFormat
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import farayan.sabad.core.base.SabadEntityBase
import farayan.sabad.core.model.product.ProductEntity

@DatabaseTable(tableName = "product-barcodes")
class ProductBarcodeEntity : SabadEntityBase<ProductBarcodeEntity>() {
    @DatabaseField(columnName = "bitmap-result-points")
    var bitmapResultPoints: String? = null

    @DatabaseField(columnName = "bitmap-scale-factor")
    var bitmapScaleFactor: Int = 0

    @DatabaseField(columnName = "bitmap-file")
    var bitmapFile: String? = null

    @DatabaseField(columnName = "value")
    var value: String? = null

    @DatabaseField(columnName = "format")
    var format: BarcodeFormat? = null

    @DatabaseField(columnName = "product", foreign = true)
    var product: ProductEntity? = null
    override fun NeedsRefresh(): Boolean {
        return false
    }
}